torch/autograd/__init__.py

```diff
@@ -28,6 +28,12 @@ def _make_grads(outputs, grads):
                                    + str(grad.shape) + " and output["
                                    + str(outputs.index(out)) + "] has a shape of "
                                    + str(out.shape) + ".")
+            if (out.dtype.is_complex and not grad.dtype.is_complex) or (not out.dtype.is_complex and grad.dtype.is_complex):
```

Comment:
I'm still not sure how this is different from `out.dtype.is_complex != grad.dtype.is_complex` ?

Reply:
updated